### DISCUSSION TITLE
Taskbar position option and "never combine"

### DISCUSSION THREAD

**Discussion by Shassk** [valinet/ExplorerPatcher#508] [Ideas]
So I'm using Windows with taskbar on top for almost a decade at this point. And luckily redegit allows to do this in Win11 as well.
However.
It doesn't work with your Win10 taskbar which is always positioned on the bottom.
Which sucks since bringing back "never combine windows" is great and all, but it works only in Win10 taskbar.
Can we see a possibility of combining those options?

**Comment by valinet** [comment]
So you want what? To move the Windows 10 taskbar to the top? Well, it’s not “stuck to the bottom”. I am a bit surprised that in 10 years you haven’t stumbled upon this: right click the taskbar, uncheck “Lock all taskbars”/“Lock taskbar”, drag it to the top and… profit! Never combine (aka show labels) can be set via the Properties GUI as well.

Btw, this question is asked at least once every week, I think I will mention it in the README…

Thanks

**Reply by Shassk** [reply]
> So you want what? To move the Windows 10 taskbar to the top? right click the taskbar, uncheck “Lock all taskbars”/“Lock taskbar”, drag it to the top and… profit!

Yes. Thanks, I'll try it. Maybe I was too confused by the fact the registry key didn't work on it, but now thinking about this it kinda makes sense.

> I am a bit surprised that in 10 years you haven’t stumbled upon this

Actually no, I think the last time I used it in WinXP times. After that only through taskbar right-click settings.

> I think I will mention it in the README…

Well maybe add a visible setting for this then? Your app gets this parameter in some way anyway, you'll just display it.

**Reply by valinet** [reply]
My app does not get the parameter. I haven’t reimplemented all the shell, this app also activates dormant functionality from explorer.exe, like the Windows 10 taskbar. That reads the taskbar placement flag and uses it (StuckRects3), not this app directly. To have it display in the interface, I need to implement some more logic to read/write that registry key, which has its own format and contains more than merely the taskbar position. Not a high priority.

**Reply by Shassk** [reply]
Oh I see, that makes sense then.
